packages/venia-concept/src/RootComponents/Category/category.js

```diff
@@ -1,22 +1,43 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { number, shape, string } from 'prop-types';
 import { usePagination, useQuery } from '@magento/peregrine';
 
 import { toggleDrawer } from 'src/actions/app';
 import catalogActions from 'src/actions/catalog';
 import { mergeClasses } from 'src/classify';
 import { loadingIndicator } from 'src/components/LoadingIndicator';
-import { connect } from 'src/drivers';
+import { connect, withRouter } from 'src/drivers';
+import { compose } from 'redux';
+
 import categoryQuery from 'src/queries/getCategory.graphql';
 import isObjectEmpty from 'src/util/isObjectEmpty';
 import { getFilterParams } from 'src/util/getFilterParamsFromUrl';
 import CategoryContent from './categoryContent';
 import defaultClasses from './category.css';
+import getQueryParameterValue from 'src/util/getQueryParameterValue';
 
 const Category = props => {
-    const { filterClear, id, openDrawer, pageSize } = props;
+    const { filterClear, id, location, openDrawer, pageSize } = props;
+    const classes = mergeClasses(defaultClasses, props.classes);
 
-    const [paginationValues, paginationApi] = usePagination();
+    // Check the query param for the initial page to provide to the pagination
+    // hook. Without this we may load `page=4` but `usePagination` will init to
+    // 1, so it'll fetch page 1 _and_ page 4 (bad!).
+    const initialPage = useMemo(
+        () =>
+            Math.max(
+                1,
+                Math.floor(
+                    ~~getQueryParameterValue({
+                        location,
+                        queryParameter: 'page'
```

Comment:
One thing that stands out to me in all of this is that I don't know who the `page` parameter "belongs" to anymore. We need to read it to get the `initialPage` value for the `usePagination` hook we initialize here in `category.js`, but the only place in which we update the query parameter is within `pagination.js`, a child component.

Reply:
That's a really good observation, related to the question in https://github.com/magento-research/pwa-studio/issues/1305. I think we need to encapsulate the query parameter handling entirely inside `usePagination`; it should use [useSearchParam](https://github.com/magento-research/pwa-studio/blob/develop/packages/peregrine/src/hooks/useSearchParam.js) under the hood.

Reply:
I made these changes. I'll wait for feedback before fixing the tests.